models/articles.model.js

```diff
@@ -7,12 +7,16 @@ exports.selectArticleById = async (targetArticleId) => {
     if(!Number.isInteger(Number(targetArticleId))){
         return Promise.reject({status: 400, msg:'Bad request!'})
     } else{
-        const insertQuery = 'SELECT * FROM articles WHERE article_id = $1;'
-        const {rows: article} = await db.query(insertQuery,[targetArticleId]);
+        const articleQuery = 'SELECT * FROM articles WHERE article_id = $1;'
+        const {rows: article} = await db.query(articleQuery,[targetArticleId]);
+        const commentQuery = 'SELECT * FROM comments WHERE article_id = $1;'
+        const {rows: comments} = await db.query(commentQuery,[targetArticleId]);
         if(article.length === 0){
             return Promise.reject({status:404, msg:'Article not found!'});
+        } else {
+            article[0].comment_count = comments.length;
+            return article;
```

Comment:
this can be done in a single SQL query instead of two separate ones, which is more efficient but this way is still okay but i'd advise if you have time to refactor.